lxd/db/generate/db/stmt.go

```diff
@@ -268,7 +273,14 @@ func (s *Stmt) create(buf *file.Buffer, replace bool) error {
 
 	for i, field := range fields {
 		if field.IsScalar() {
-			ref := lex.Snake(field.Name)
+			joinTable, _, err := field.ScalarTableColumn()
+			if err != nil {
+				return err
+			}
+
+			// Assumes that the reference column for table "things" is of the form "thing_id"
+			ref := lex.Singular(joinTable)
```

Comment:
Do we still need this if its handled by https://github.com/lxc/lxd/pull/10841

Reply:
I'm honestly not sure. It depends if we want to be prescriptive about how the joins are made (i.e. every db tag containing `join`/`leftjoin` must have an accompanying `joinon`). Otherwise we can use `joinon` if it exists and fallback to making this assumption.

Reply:
My vote would be for being prescriptive and removing the magic :)

Reply:
Sure, I can update this once #10841 is merged. FYI that will require adding `joinon` to many LXD structs e.g. profiles will become:
```go
type Profile struct {
	ID          int
	ProjectID   int    `db:"omit=create,update"`
	Project     string `db:"primary=yes&join=projects.name&joinon=project_id"`
	Name        string `db:"primary=yes"`
	Description string `db:"coalesce=''"`
}
```

Reply:
That sounds good thanks

Reply:
Note removing this logic that this would currently break snapshots, and it would need its own manual query I believe.